uboot-mediatek: add support for Zyxel WX5600-T0 with custom partitions

ftp21 opened the pull request:
This access point is a ‘friend’ of the T56 supplied by Odido but with DDR3 RAM and with two Ethernet 2.5 (GPY211)

The flash procedure is similar to other Zyxel T56/EX5600/EX5601

If you need backup please use the T56 guide

We can install firmware with mtk_uartboot or with supervisor password method described on EX5601 thread.

Specifications:
SOC: MT7986b
RAM: 512MB
Flash: 512 MB SPI NAND
Ports: 2 LAN 2.5Gbps (GPY211C)
WIFI: MT7976GN + MT7976AN
LED: 3 bicolor LED - 1 monocolor LED
Buttons: Reset and WPS


mtk_uartboot method:
Load Uboot:

```
./mtk_uartboot -a -p ./mt7986-ram-ddr3-bl2.bin -s /dev/ttyUSB0 -f openwrt-mediatek-filogic-zyxel_wx5600-t0-ubootmod-bl31-uboot.fip
```
**WARNING: Please use a GBIT ethernet or force it on system**
**WARNING: Please use only LAN2 port in Uboot**
Press 0 on Bootmenu
```
mtd erase ubi
ubi part ubi
bootmenu
```
Load and write BL2 and U-boot:
```
8
7
```

Load and write recovery and production
```
6
5
```

abajk commented on target/linux/mediatek/filogic/base-files/etc/board.d/01_leds:
```diff
@@ -188,11 +188,11 @@ zyxel,nwa50ax-pro)
 	ucidef_set_led_netdev "uplink" "UPLINK" "mdio-bus:05:amber:wan" "eth0" "link_10 link_100 link_2500 tx rx"
 	ucidef_set_led_netdev "uplink" "UPLINK" "mdio-bus:05:green:wan" "eth0" "link_1000 link_2500 tx rx"
 	;;
-zyxel,ex5700-telenor)
```
The entry for EX5700 should be restored.

ftp21 replied:
Fixed

ftp21 commented:
thanks a lot i updated the guide.
Yesterday i try also this method and it work well
[https://forum.openwrt.org/t/openwrt-support-for-wx5600/220626/20](https://forum.openwrt.org/t/openwrt-support-for-wx5600/220626/20)

1980490718 commented:
`target/linux/mediatek/filogic/base-files/etc/board.d/02_network`
-	huasifei,wh3000\|
+	huasifei,wh3000|\
+ 	zyxel,wx5600-t0-ubootmod)

ftp21 commented:
> All changes Sort from a-z.

Hi 
I have very bad eyesight :) 
Would you show me where to order the changes?

testuser7 commented on package/boot/arm-trusted-firmware-mediatek/Makefile:
```diff
@@ -352,6 +352,15 @@ define Trusted-Firmware-A/mt7986-spim-nand-ddr3
   DDR_TYPE:=ddr3
 endef
 
+define Trusted-Firmware-A/mt7986-spim-nand-4k-ddr3
```
`4k` should come before `ddr3`.

testuser7 commented on package/boot/arm-trusted-firmware-mediatek/Makefile:
```diff
@@ -561,6 +570,7 @@ TFA_TARGETS:= \
 	mt7986-snand-ddr3 \
 	mt7986-spim-nand-ddr3 \
 	mt7986-spim-nand-ubi-ddr3 \
+	mt7986-spim-nand-4k-ddr3 \
```
Likewise, `4k` should come before `ddr3`.

testuser7 commented on target/linux/mediatek/image/filogic.mk:
```diff
@@ -2003,3 +2003,32 @@ define Device/zyxel_nwa50ax-pro
+  DEVICE_PACKAGES := kmod-mt7915e kmod-mt7986-firmware mt7986-wo-firmware
+  KERNEL_INITRAMFS_SUFFIX := -recovery.itb
+  IMAGES := sysupgrade.itb
+  UBINIZE_OPTS := -E 5
```
Is this option necessary?

ftp21 replied:
I copied the EX5601 and EX5700 options exactly, I don't know what “eof-markers to put at the end of the output image” is, if the option is not needed I can delete it

testuser7 replied:
If I remember correctly, in some devices you need to overwrite more data and that's why this option was added.

ftp21 replied:
This option is also in the original sources https://github.com/pameruoso/zyxel-ex5601t0/blob/e3077d117a9bdaf3af60da4fb88f119b05157523/target/linux/mediatek/image/wx5600t0.mk#L5

testuser7 replied:
But that doesn't mean this option is needed...

testuser7 commented on package/boot/arm-trusted-firmware-mediatek/Makefile:
```diff
@@ -561,6 +570,7 @@ TFA_TARGETS:= \
 	mt7986-snand-ddr3 \
 	mt7986-spim-nand-ddr3 \
 	mt7986-spim-nand-ubi-ddr3 \
+	mt7986-spim-nand-4k-ddr3 \
```
```suggestion
	mt7986-spim-nand-4k-ddr3 \
	mt7986-spim-nand-ddr3 \
	mt7986-spim-nand-ubi-ddr3 \
```

ftp21 replied:
Also 4k-ddr4 use this sorting, is 4k-ddr4 wrong?

testuser7 commented on package/boot/arm-trusted-firmware-mediatek/Makefile:
```diff
@@ -361,6 +361,15 @@ define Trusted-Firmware-A/mt7986-spim-nand-ubi-ddr3
   USE_UBI:=1
 endef
 
+define Trusted-Firmware-A/mt7986-spim-nand-4k-ddr3
```
Move that definition before `Trusted-Firmware-A/mt7986-spim-nand-ddr3`.

abajk commented on target/linux/mediatek/dts/mt7986b-zyxel-wx5600-t0-ubootmod.dts:
```diff
@@ -0,0 +1,443 @@
+                #size-cells = <1>;
+                compatible = "spi-nand";
+                reg = <1>;
+                spi-max-frequency = <20000000>;
```
Can it go higher?

On the EX5601, 52 MHz frequency works well. You can also include the specifications of the device in the commit description. It will then be clear what memory chips are mounted in this device.

ftp21 commented on target/linux/mediatek/dts/mt7986b-zyxel-wx5600-t0-ubootmod.dts:
```diff
@@ -0,0 +1,444 @@
+                #size-cells = <1>;
+                compatible = "spi-nand";
+                reg = <1>;
+                spi-max-frequency = <50000000>;
```
fixed

testuser7 commented:
The commit changes are a bit mixed up. You add the main dts in the commit regarding uboot.
You can also add an additional newline character after the commit title in message.

testuser7 commented:
> OK, could you please highlight the exact points where you think there are errors?

These 3 files should be in the uboot commit:
- package/boot/arm-trusted-firmware-mediatek/Makefile
- package/boot/uboot-mediatek/Makefile
- package/boot/uboot-mediatek/patches/461-add-zyzel-wx5600-t0.patch

testuser7 commented on package/boot/uboot-mediatek/patches/461-add-zyzel-wx5600-t0.patch:
```diff
@@ -0,0 +1,426 @@
++reset_factory=ubi part ubi ; mw $loadaddr 0x0 0x800 ; ubi write $loadaddr ubootenv 0x800 ; ubi write $loadaddr ubootenv2 0x800
++mtd_write_fip=mtd erase fip && mtd write fip $loadaddr
++mtd_write_bl2=mtd erase bl2 && mtd write bl2 $loadaddr
++ubi_create_env=ubi check ubootenv || ubi create ubootenv 0x100000 dynamic ; ubi check ubootenv2 || ubi create ubootenv2 0x100000 dynamic
```
Isn't the size of `ubootenv` volumes too large? For OpenWrt One size `0x1f000` is used: https://github.com/testuser7/openwrt/blob/mediatek-filogic-main-da50c4a/package/boot/uboot-mediatek/patches/453-add-openwrt-one.patch#L565

testuser7 replied:
Perhaps this needs to be clarified, as there is no point in allocating so much space to variables.

ftp21 commented:
> > OK, could you please highlight the exact points where you think there are errors?
> 
> These 3 files should be in the uboot commit:
> 
> * package/boot/arm-trusted-firmware-mediatek/Makefile
> * package/boot/uboot-mediatek/Makefile
> * package/boot/uboot-mediatek/patches/461-add-zyzel-wx5600-t0.patch

Fixed in this rebase, thanks

testuser7 commented on target/linux/mediatek/dts/mt7986b-zyxel-wx5600-t0-ubootmod.dts:
```diff
@@ -0,0 +1,444 @@
+		read-only;
+	};
+
+	partition@540000 {
```
Do we need to keep partition `zloader` if it is overlapped by `fip`?
```
	  - 0x000000380000-0x000000580000 : "fip"
	  - 0x000000540000-0x000000580000 : "zloader"
```

ftp21 replied:
Ops sorry fixed according original DTS https://github.com/pameruoso/zyxel-ex5601t0/blob/e3077d117a9bdaf3af60da4fb88f119b05157523/target/linux/mediatek/files-5.4/arch/arm64/boot/dts/mediatek/mt7986-spim-nand-512M-256kB-partition.dtsi

testuser7 replied:
That's a good idea, I did something similar here: https://github.com/openwrt/openwrt/commit/c9b324d3ac084253171ac6233ca12cb4feacf36c

ftp21 replied:
Ok can you help me with refactor?

Djfe commented:
Hi,
thank you for adding support for this great device.
Have you thought about adding device specs to your commit message? :)
It might be related to the t56 but it still has differences/is not the same device.

testuser7 commented on target/linux/mediatek/dts/mt7986b-zyxel-wx5600-t0-ubootmod.dts:
```diff
@@ -0,0 +1,444 @@
+// SPDX-License-Identifier: (GPL-2.0 OR MIT)
+
+/dts-v1/;
+#include <dt-bindings/input/input.h>
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/leds/common.h>
+
+#include "mt7986b.dtsi"
```
```suggestion
#include "mt7986b.dtsi"
#include <dt-bindings/gpio/gpio.h>
#include <dt-bindings/input/input.h>
#include <dt-bindings/leds/common.h>
#include <dt-bindings/pinctrl/mt65xx.h>
```

testuser7 commented on target/linux/mediatek/dts/mt7986b-zyxel-wx5600-t0-ubootmod.dts:
```diff
@@ -0,0 +1,444 @@
+		led-failsafe = &led_red_wps;
+		led-running = &led_green_pwr;
+		led-upgrade = &led_green_wps;
+
```
```suggestion
```

testuser7 commented on target/linux/mediatek/dts/mt7986b-zyxel-wx5600-t0-ubootmod.dts:
```diff
@@ -0,0 +1,444 @@
+/ {
+	model = "Zyxel WX5600-T0 ubootmod";
+	compatible = "zyxel,wx5600-t0-ubootmod", "mediatek,mt7986b";
+	
```
```suggestion

```

testuser7 commented on target/linux/mediatek/dts/mt7986b-zyxel-wx5600-t0-ubootmod.dts:
```diff
@@ -0,0 +1,444 @@
+
+	memory@40000000 {
+		device_type = "memory";
+		reg = <0x40000000 0x10000000>;
```
```suggestion
		reg = <0 0x40000000 0 0x10000000>;
```

testuser7 commented on target/linux/mediatek/dts/mt7986b-zyxel-wx5600-t0-ubootmod.dts:
```diff
@@ -0,0 +1,444 @@
+};
+
+&spi0 {
+        pinctrl-names = "default";
+        pinctrl-0 = <&spi_flash_pins>;
+        cs-gpios = <0>, <0>;
+        #address-cells = <1>;
+        #size-cells = <0>;
+        status = "okay";
+
+        spi_nand: spi_nand@0 {
+                #address-cells = <1>;
+                #size-cells = <1>;
+                compatible = "spi-nand";
+                reg = <1>;
+                spi-max-frequency = <52000000>;
+                spi-tx-bus-width = <4>;
+                spi-rx-bus-width = <4>;
```
```suggestion
	pinctrl-names = "default";
	pinctrl-0 = <&spi_flash_pins>;
	cs-gpios = <0>, <0>;
	#address-cells = <1>;
	#size-cells = <0>;
	status = "okay";

	spi_nand: spi_nand@0 {
		#address-cells = <1>;
		#size-cells = <1>;
		compatible = "spi-nand";
		reg = <0>;
		spi-max-frequency = <52000000>;
		spi-tx-bus-width = <4>;
		spi-rx-bus-width = <4>;
```

testuser7 commented on target/linux/mediatek/dts/mt7986b-zyxel-wx5600-t0-ubootmod.dts:
```diff
@@ -0,0 +1,444 @@
+	};
+};
+
+
```
```suggestion
```

testuser7 commented on target/linux/mediatek/dts/mt7986b-zyxel-wx5600-t0-ubootmod.dts:
```diff
@@ -0,0 +1,444 @@
+	status = "okay";
+	pinctrl-names = "default", "dbdc";
+	pinctrl-0 = <&wf_2g_5g_pins>;
+	pinctrl-1 = <&wf_dbdc_pins>;
```
```suggestion
	pinctrl-1 = <&wf_dbdc_pins>;
	nvmem-cells = <&eeprom_factory>;
	nvmem-cell-names = "eeprom";
```

testuser7 commented on target/linux/mediatek/dts/mt7986b-zyxel-wx5600-t0-ubootmod.dts:
```diff
@@ -0,0 +1,444 @@
+		};
+		conf-pu {
+			pins = "SPI2_CS", "SPI2_HOLD", "SPI2_WP";
+			drive-strength = <4>;
+			mediatek,pull-up-adv = <0>;	/* bias-disable */
+		};
+		conf-pd {
+			pins = "SPI2_CLK", "SPI2_MOSI", "SPI2_MISO";
+			drive-strength = <4>;
```
```suggestion
			drive-strength = <MTK_DRIVE_8mA>;
			mediatek,pull-up-adv = <0>;	/* bias-disable */
		};
		conf-pd {
			pins = "SPI2_CLK", "SPI2_MOSI", "SPI2_MISO";
			drive-strength = <MTK_DRIVE_8mA>;
```
This change is only needed for Eltron flash. We have Micron chip:
```
[    0.553160] spi-nand spi0.1: Micron SPI NAND was found.
[    0.558381] spi-nand spi0.1: 512 MiB, block size: 256 KiB, page size: 4096, OOB size: 256
```

testuser7 commented on target/linux/mediatek/dts/mt7986b-zyxel-wx5600-t0-ubootmod.dts:
```diff
@@ -0,0 +1,444 @@
+	};
+
+	uart0_pins: uart0-pins {
+                mux {
+                        function = "uart";
+                        groups = "uart0";
+                };
+        };
```
```suggestion
		mux {
			function = "uart";
			groups = "uart0";
		};
	};
```

testuser7 commented on target/linux/mediatek/dts/mt7986b-zyxel-wx5600-t0-ubootmod.dts:
```diff
@@ -0,0 +1,444 @@
+		};
+		conf {
+			pins = "WF0_HB1", "WF0_HB2", "WF0_HB3", "WF0_HB4",
+			       "WF0_HB0", "WF0_HB0_B", "WF0_HB5", "WF0_HB6",
+			       "WF0_HB7", "WF0_HB8", "WF0_HB9", "WF0_HB10",
+			       "WF0_TOP_CLK", "WF0_TOP_DATA", "WF1_HB1",
+			       "WF1_HB2", "WF1_HB3", "WF1_HB4", "WF1_HB0",
+			       "WF1_HB5", "WF1_HB6", "WF1_HB7", "WF1_HB8",
+			       "WF1_TOP_CLK", "WF1_TOP_DATA";
+			drive-strength = <4>;
```
```suggestion
				"WF0_HB0", "WF0_HB0_B", "WF0_HB5", "WF0_HB6",
				"WF0_HB7", "WF0_HB8", "WF0_HB9", "WF0_HB10",
				"WF0_TOP_CLK", "WF0_TOP_DATA", "WF1_HB1",
				"WF1_HB2", "WF1_HB3", "WF1_HB4", "WF1_HB0",
				"WF1_HB5", "WF1_HB6", "WF1_HB7", "WF1_HB8",
				"WF1_TOP_CLK", "WF1_TOP_DATA";
			drive-strength = <MTK_DRIVE_4mA>;
```

testuser7 commented on target/linux/mediatek/dts/mt7986b-zyxel-wx5600-t0-ubootmod.dts:
```diff
@@ -0,0 +1,444 @@
+		};
+		conf {
+			pins = "WF0_HB1", "WF0_HB2", "WF0_HB3", "WF0_HB4",
+			       "WF0_HB0", "WF0_HB0_B", "WF0_HB5", "WF0_HB6",
+			       "WF0_HB7", "WF0_HB8", "WF0_HB9", "WF0_HB10",
+			       "WF0_TOP_CLK", "WF0_TOP_DATA", "WF1_HB1",
+			       "WF1_HB2", "WF1_HB3", "WF1_HB4", "WF1_HB0",
+			       "WF1_HB5", "WF1_HB6", "WF1_HB7", "WF1_HB8",
+			       "WF1_TOP_CLK", "WF1_TOP_DATA";
+			drive-strength = <4>;
```
```suggestion
				"WF0_HB0", "WF0_HB0_B", "WF0_HB5", "WF0_HB6",
				"WF0_HB7", "WF0_HB8", "WF0_HB9", "WF0_HB10",
				"WF0_TOP_CLK", "WF0_TOP_DATA", "WF1_HB1",
				"WF1_HB2", "WF1_HB3", "WF1_HB4", "WF1_HB0",
				"WF1_HB5", "WF1_HB6", "WF1_HB7", "WF1_HB8",
				"WF1_TOP_CLK", "WF1_TOP_DATA";
			drive-strength = <MTK_DRIVE_4mA>;
```

testuser7 commented on target/linux/mediatek/dts/mt7986b-zyxel-wx5600-t0-ubootmod.dts:
```diff
@@ -0,0 +1,444 @@
+	};
+};
+
+
```
```suggestion
```

testuser7 commented on target/linux/mediatek/dts/mt7986b-zyxel-wx5600-t0-ubootmod.dts:
```diff
@@ -0,0 +1,444 @@
+		phy-mode = "2500base-x";
+		phy-handle = <&phy5>;
+		label = "lan2";
+	
```
```suggestion
```

testuser7 commented on target/linux/mediatek/dts/mt7986b-zyxel-wx5600-t0-ubootmod.dts:
```diff
@@ -0,0 +1,444 @@
+		#address-cells = <1>;
+		#size-cells = <0>;
+		
+		
```
```suggestion
```

testuser7 commented on target/linux/mediatek/dts/mt7986b-zyxel-wx5600-t0-ubootmod.dts:
```diff
@@ -0,0 +1,444 @@
+			reset-assert-us = <600>;
+			reset-deassert-us = <20000>;
+			reg = <5>;
+			
```
```suggestion
```

testuser7 commented on target/linux/mediatek/dts/mt7986b-zyxel-wx5600-t0-ubootmod.dts:
```diff
@@ -0,0 +1,444 @@
+};
+
+&gmac0 {
+	nvmem-cells = <&macaddr_factory_0024 0>;
+	nvmem-cell-names = "mac-address";
+};
+
+&gmac1 {
+	nvmem-cells = <&macaddr_factory_002a 0>;
```
```suggestion
	nvmem-cells = <&macaddr_factory_24 0>;
	nvmem-cell-names = "mac-address";
};

&gmac1 {
	nvmem-cells = <&macaddr_factory_2a 0>;
```

testuser7 commented on target/linux/mediatek/dts/mt7986b-zyxel-wx5600-t0-ubootmod.dts:
```diff
@@ -0,0 +1,444 @@
+	nvmem-cell-names = "mac-address";
+};
+
+&wifi {
+	nvmem-cells = <&eeprom_factory>;
+	nvmem-cell-names = "eeprom";
+};
```
```suggestion
```

testuser7 reviewed (changes requested):
Some dts cleanup and fixes

testuser7 reviewed (approved):
Tested-by: Paweł Owoc <frut3k7@gmail.com>

@hauke Can you look at this PR?

DragonBluep commented:
Please fix the CI warning first https://github.com/openwrt/openwrt/actions/runs/19628283099/job/56201743107?pr=18364.

DragonBluep commented on package/boot/uboot-mediatek/patches/461-add-zyzel-wx5600-t0.patch:
```diff
@@ -0,0 +1,427 @@
++CONFIG_BOOTDELAY=30
++CONFIG_AUTOBOOT_KEYED=y
++CONFIG_AUTOBOOT_MENU_SHOW=y
++CONFIG_DEFAULT_FDT_FILE="mediatek/mt7986b-zyxel_wx5600-t0.dtb"
```
```suggestion
```

DragonBluep commented on package/boot/arm-trusted-firmware-mediatek/Makefile:
```diff
@@ -394,6 +394,15 @@ define Trusted-Firmware-A/mt7986-spim-nand-ubi-ddr3
 endef
 
+define Trusted-Firmware-A/mt7986-spim-nand-4k-ddr3
+  NAME:=MediaTek MT7986 (SPI-NAND via SPIM, DDR3)
```
Something like `MediaTek MT7986 (SPI-NAND 4K+256 via SPIM, DDR3)`. `MediaTek MT7986 (SPI-NAND via SPIM, DDR3)` was already used by other package.

ftp21 replied:
Thanks a lot i changed also DDR4 4k+256

DragonBluep commented on target/linux/mediatek/dts/mt7986b-zyxel-wx5600-t0-ubootmod.dts:
```diff
@@ -0,0 +1,435 @@
+			panic-indicator;
+		};
+
+		led_green_signal: green_signal {
```
No. Just remove `led_green_signal: `. This label is not referenced by other nodes.

ftp21 commented:
It should be fine now. Someone can check now?

ftp21 commented on target/linux/mediatek/dts/mt7986b-zyxel-wx5600-t0-ubootmod.dts:
```diff
@@ -0,0 +1,435 @@
+
+	spi_nand: spi_nand@0 {
+		#address-cells = <1>;
+		#size-cells = <1>;
```
Unfortunately, using two different workstations also leads to these oversights :)

testuser7 commented on package/boot/uboot-mediatek/patches/471-add-zyzel-wx5600-t0.patch:
```diff
@@ -0,0 +1,402 @@
++CONFIG_OF_EMBED=y
++CONFIG_ENV_OVERWRITE=y
++CONFIG_ENV_IS_IN_UBI=y
++CONFIG_SYS_REDUNDAND=y
```
```suggestion
+CONFIG_ENV_REDUNDANT=y
```
This env is renamed: https://github.com/u-boot/u-boot/commit/5fb88fa725d6d4f6d358e86613a14cde01426692

ftp21 committed:
uboot-mediatek: add support for Zyxel WX5600-T0

This access point is a ‘friend’ of the T56 supplied by Odido but with DDR3 RAM and with two Ethernet 2.5 (GPY211)

The flash procedure is similar to other Zyxel T56/EX5600/EX5601

If you need backup please use the T56 guide

Please refer to https://openwrt.org/inbox/toh/zyxel/wx5600-t0 for detailed flash informations

We can install all with U-boot and mtk_uartboot.
Load Uboot:

```
./mtk_uartboot -a -p ./mt7986-ram-ddr3-bl2.bin -s /dev/ttyUSB0 -f openwrt-mediatek-filogic-zyxel_wx5600-t0-ubootmod-bl31-uboot.fip
```
**WARNING: Please use a GBIT ethernet or force it on system**
**WARNING: Please use only LAN2 port in Uboot**
Press 0 on Bootmenu
```
mtd erase ubi
run ubi_format
bootmenu
```
Load and write BL2 and U-boot:
```
8
7
```

Load and write recovery and production
```
6
5
```

Signed-off-by: Valerio 'ftp21' Mancini <ftp21@ftp21.eu>
Link: https://github.com/openwrt/openwrt/pull/18364
Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>

ftp21 committed:
mediatek: add support for Zyxel WX5600-T0

This access point is a ‘friend’ of the T56 supplied by Odido but with DDR3 RAM and with two Ethernet 2.5 (GPY211)

The flash procedure is similar to other Zyxel T56/EX5600/EX5601

If you need backup please use the T56 guide
Please refer to https://openwrt.org/inbox/toh/zyxel/wx5600-t0 for detailed flash informations

Specifications:
SOC: MT7986b
RAM: 512MB
Flash: 512 MB SPI NAND
Ports: 2 LAN 2.5Gbps (GPY211C)
WIFI: MT7976GN + MT7976AN
LED: 3 bicolor LED - 1 monocolor LED
Buttons: Reset and WPS

We can install all with U-boot and mtk_uartboot.
Load Uboot:

```
./mtk_uartboot -a -p ./mt7986-ram-ddr3-bl2.bin -s /dev/ttyUSB0 -f openwrt-mediatek-filogic-zyxel_wx5600-t0-ubootmod-bl31-uboot.fip
```
**WARNING: Please use a GBIT ethernet or force it on system**
**WARNING: Please use only LAN2 port in Uboot**
Press 0 on Bootmenu
```
mtd erase ubi
run ubi_format
bootmenu
```
Load and write BL2 and U-boot:
```
8
7
```

Load and write recovery and production
```
6
5
```

Signed-off-by: Valerio 'ftp21' Mancini <ftp21@ftp21.eu>
Co-authored-by: Hal Martin <halmartin@gmail.com>
Link: https://github.com/openwrt/openwrt/pull/18364
Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>
